### PR TITLE
Improvement to the maskedLinkPaste plugin

### PR DESCRIPTION
**Pull request by NeoNyaa**
Added a setting which allows the user to disable link embedding for their masked pastes by wrapping the link inbetween the `<>` symbols

### CODE REVIEW

**Comment by Vendicated** [comment]
hmm i feel like it would be more useful if it worked via modifier key?

normal paste -> embed
shift + paste -> suppress embed

**Comment by NeoNyaa** [comment]
> > hmm i feel like it would be more useful if it worked via modifier key?
> > normal paste -> embed shift + paste -> suppress embed
> 
> I'd argue otherwise purely because the point of masked links is to keep messages with links neat and tidy, another issue with embeds is they strip URL parameters so things like game invits break often.
> 
> Having it be a toggle without the need to restart vencord worked well and I recon most people would opt to have links not embed when masking them.

To expand on this, from a UX standpoint, it would be better to have the user choose either embedding masked links or non embedding masked links by default. If their usual default doesn't work for a random use case, they can manually remove the `<>` wrapping the link. Using keyboard shortcuts can become a little iffy when many different operating systems have different defaults. While it may work on Windows to use `CTRL + V` or `CTRL + SHIFT + V` who's to say that it'll work on Linux for example which can have some very random shortcuts.

I'd much prefer setting a default and manually removing the `<>` wrapper than fighting with system shortcuts for whatever reason, especially if the shortcut is used often for the OS and some random Vencord plugin is the only thing causing issues.

File: src/plugins/maskedLinkPaste/index.ts [hunk 1]
```diff
@@ -28,10 +39,17 @@ export default definePlugin({
...
+                SlateTransforms.insertText(
+                    editor,
+                    `[${editor.operations[0].text}](<${content}>)`
+                );
+            } else {
+                SlateTransforms.insertText(
+                    editor,
+                    `[${editor.operations[0].text}](${content})`
+                );
+            }
```
Comment (Vendicated) [inline src/plugins/maskedLinkPaste/index.ts:52]:
```suggestion
			const link = settings.store.disableEmbed ? `<${content}>` : content;
            SlateTransforms.insertText(
                editor,
                `[${editor.operations[0].text}](${link})`
            );
```

File: src/plugins/maskedLinkPaste/index.ts [hunk 2]
```diff
@@ -4,18 +4,29 @@
...
+        default: true,
+        description: "Wrap the link inbetween < > to prevent it from embedding.",
+        restartNeeded: false
+    }
+});
+
 export default definePlugin({
     name: "MaskedLinkPaste",
-    authors: [Devs.TheSun],
+    authors: [Devs.TheSun, Devs.Neo],
```
Comment (Vendicated) [inline src/plugins/maskedLinkPaste/index.ts:27]: please don't add yourself to authors for such a minor change

Comment (NeoNyaa) [reply src/plugins/maskedLinkPaste/index.ts:27]: I was following the documentation for what to do :/

Comment (Vendicated) [reply src/plugins/maskedLinkPaste/index.ts:27]: that's okay, i wasn't blaming you or anything!

**Comment by Nuckyz** [comment]
This plugin is a stock Discord feature now.

**Pull request closed without merging**